src/applications/disability-benefits/all-claims/utils/validations.js

```diff
@@ -0,0 +1,241 @@
+import moment from 'moment';
+import { convertToDateField } from '~/platform/forms-system/src/js/validation';
+import { isValidDateRange } from '~/platform/forms/validations';
+
+/* The particular field used for the Toxic Exposure date inputs properly executes
+* validateCurrentOrPastDate, however schema validations against the individual fields
+* (day, month, year) do not properly block the Veteran from progressing.
+* These validations are necessary to properly block progression and prevent what
+* we're calling the 'XX' date issue.
+*/
+export function validateToxicExposureGulfWar1990Dates(
+  errors,
+  { startDate, endDate },
+) {
+  // With these fields in particular, a missing value will convert to
+  // a date format with 'XX' e.g. 01-XX-1990.
+  // Calling convertToDateField on a partial date will return an empty date object.
+  // We will validate around this fact.
+  const startDateApproximate = convertToDateField(startDate);
+  const endDateApproximate = convertToDateField(endDate);
+
+  const messages = {
+    range: 'Your service start date must be before your end date',
+    enddate: 'Service end date must be after August 2, 1990',
+    // This is largely redundant because the component validation already prevents
+    // inputting dates in the future. However, this is in place just in case, because...
+    // TODO: The current schema validation says that the Veteran can input a year
+    // up to 100 years in the future. That is incorrect and should be corrected/removed.
+    futuredate: 'Service date can not be in the future',
+    // Schema validations will display an error when a value is missing, however it
+    // will not properly prevent the Veteran from progressing, necessitating this.
+    // TODO: The error message here can be made more specific, but that will somewhat
+    // depend on what decisions (and schema updates) are made in regards to partial dates.
+    missingvalue: 'Service date missing month, day, or year',
+  };
+
+  // Range validations
+  if (!isValidDateRange(startDateApproximate, endDateApproximate)) {
+    errors.endDate.addError(messages.range);
+  }
+
+  if (
+    endDateApproximate.year.value < 1990 ||
+    (endDateApproximate.year.value === 1990 &&
+      endDateApproximate.day.value < 8 &&
+      endDateApproximate.month.value < 2)
```

Comment:
I believe the `month` value should be **8** and the `day` value should be **2**.

Also, the logic can be improved, as a date of July 31, 1990, for example, would evaluate to **false**, when the expectation, it seems, is for anything on or before August 2, 1990 to be **true**.